src/network/networkd-mptcp.c

```diff
@@ -0,0 +1,398 @@
+        if (r < 0)
+                return log_oom();
+
+        if (isempty(rvalue)) {
+                mp_tcp->id_is_set = false;
+
+                TAKE_PTR(mp_tcp);
+                return 0;
+        }
+
+        r = safe_atou8(rvalue, &mp_tcp->id);
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to parse MPTCP '%s=', ignoring assignment: %s", lvalue, rvalue);
+                return 0;
+        }
+
+        mp_tcp->id_is_set = true;
+
+        TAKE_PTR(mp_tcp);
+        return 0;
+}
+
+int config_parse_mp_tcp_address(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(mp_tcp_free_or_set_invalidp) MPTCP *mp_tcp = NULL;
+        Network *network = data;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = mp_tcp_new_static(network, filename, section_line, &mp_tcp);
+        if (r < 0)
+                return log_oom();
+
+        if (isempty(rvalue)) {
+                mp_tcp->address = IN_ADDR_NULL;
+                mp_tcp->family = AF_UNSPEC;
+
+                TAKE_PTR(mp_tcp);
+                return 0;
+        }
+
+        r = in_addr_from_string_auto(rvalue, &mp_tcp->family, &mp_tcp->address);
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to parse MPTCP '%s=', ignoring assignment: %s", lvalue, rvalue);
+                return 0;
+        }
+
+        TAKE_PTR(mp_tcp);
+        return 0;
+}
+
+int config_parse_mp_tcp_uint32(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(mp_tcp_free_or_set_invalidp) MPTCP *mp_tcp = NULL;
+        uint32_t *v = data;
+        uint32_t k;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                *v = UINT32_MAX;
+
+                TAKE_PTR(mp_tcp);
+                return 0;
+        }
+
+        r = safe_atou32(rvalue, &k);
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to parse MPTCP '%s=', ignoring assignment: %s", lvalue, rvalue);
+                return 0;
+        }
+        if (k > 8) {
```

Comment:
your man page addition says valid range starts with 1, but this here allows 0, too